nemo_skills/dataset/wmt24pp/prepare.py

```
@@ -12,49 +12,169 @@
+                    continue
+                doc_id = row["document_id"]
+                if doc_id not in docs:
+                    docs[doc_id] = []
+                docs[doc_id].append(row)
+
+            for doc_id in docs:
+                docs[doc_id].sort(key=lambda r: r["segment_id"])
+
+            source_sentences = []
+            reference_sentences = []
+            for doc_id, rows in docs.items():
```

Ruff (0.15.9)

[warning] Loop control variable `doc_id` not used within loop body

(B007)